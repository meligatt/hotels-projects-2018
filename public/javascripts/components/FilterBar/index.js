import React from 'react';
import PropTypes from 'prop-types';

const FilterBar = ({
  optionSelected,
  onOptionChange
}) => {
  return(
    <div>
    Sort by
    <select value={ optionSelected } onChange = { (e) => {onOptionChange(e)} } >
    <option value="desc"> Price (high-low)</option>
    <option value="asc"> Price (low-high)</option>
    </select>
    </div>
  )
}

FilterBar.propTypes = {
  optionSelected: PropTypes.string.isRequired,
  onOptionChange: PropTypes.func.isRequired,
}

export default FilterBar;
