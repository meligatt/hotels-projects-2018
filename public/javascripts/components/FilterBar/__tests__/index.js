import React from 'react';
import FilterBar from  '../index.js';

import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe("FilterBar", () => {
  it("should render correctly", () => {
    const optionSelectedMock = "desc";
    const onOptionChangeMock = jest.fn();

    const wrapper =  shallow(
      <FilterBar
        optionSelected = { optionSelectedMock }
        onOptionChange = { onOptionChangeMock }
      />)
    expect(wrapper).toMatchSnapshot();
  })

  it("calls onOptionChange when select changes", () => {
    const optionSelectedMock = "desc";
    const onOptionChangeMock = jest.fn();

    const wrapper =  shallow(
      <FilterBar
        optionSelected = { optionSelectedMock }
        onOptionChange = { onOptionChangeMock }
      />);

      wrapper.find('select').simulate('change', 'VALUE');
      expect(onOptionChangeMock).toHaveBeenCalledWith('VALUE');
  })
})
