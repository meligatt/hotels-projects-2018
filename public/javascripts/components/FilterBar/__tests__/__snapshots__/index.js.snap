// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBar
    onOptionChange={[MockFunction]}
    optionSelected="desc"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Sort by",
        <select
          onChange={[Function]}
          value="desc"
        >
          <option
            value="desc"
          >
             Price (high-low)
          </option>
          <option
            value="asc"
          >
             Price (low-high)
          </option>
        </select>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Sort by",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="desc"
            >
               Price (high-low)
            </option>,
            <option
              value="asc"
            >
               Price (low-high)
            </option>,
          ],
          "onChange": [Function],
          "value": "desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Price (high-low)",
              "value": "desc",
            },
            "ref": null,
            "rendered": " Price (high-low)",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Price (low-high)",
              "value": "asc",
            },
            "ref": null,
            "rendered": " Price (low-high)",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Sort by",
          <select
            onChange={[Function]}
            value="desc"
          >
            <option
              value="desc"
            >
               Price (high-low)
            </option>
            <option
              value="asc"
            >
               Price (low-high)
            </option>
          </select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Sort by",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="desc"
              >
                 Price (high-low)
              </option>,
              <option
                value="asc"
              >
                 Price (low-high)
              </option>,
            ],
            "onChange": [Function],
            "value": "desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Price (high-low)",
                "value": "desc",
              },
              "ref": null,
              "rendered": " Price (high-low)",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Price (low-high)",
                "value": "asc",
              },
              "ref": null,
              "rendered": " Price (low-high)",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
